paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -0,0 +1,1175 @@
+      if (attrNamespace == null) {
+        throw XmlSyntaxException(
+          "Undefined prefix of attr resource name \"$name\"",
+          parser,
+          getDisplayName(file)
+        )
+      }
+    }
+    val formatString = parser.getAttributeValue(null, ATTR_FORMAT)
+    val formats = if (formatString.isNullOrBlank()) EnumSet.noneOf(
+      AttributeFormat::class.java
+    ) else {
+      AttributeFormat.parse(formatString)
+    }
+
+    // The average number of enum or flag values is 7 for Android framework, so start with small maps.
+    forSubTags(
+      null,
+      object : XmlTagVisitor {
+        override fun visitTag() {
+          if (parser.prefix == null) {
+            val tagName = parser.name
+            val format =
+              if (tagName == TAG_ENUM) AttributeFormat.ENUM else if (tagName == TAG_FLAG) AttributeFormat.FLAGS else null
+            if (format != null) {
+              formats.add(format)
+              val valueName = parser.getAttributeValue(null, ATTR_NAME)
+              if (valueName != null) {
+                val value = parser.getAttributeValue(null, ATTR_VALUE)
+                var numericValue: Int? = null
+                if (value != null) {
+                  try {
+                    // Integer.decode/parseInt can't deal with hex value > 0x7FFFFFFF so we use Long.decode instead.
+                    numericValue = java.lang.Long.decode(value).toInt()
+                  } catch (ignored: NumberFormatException) {
+                  }
+                }
+                resourceValue.addValue(valueName, numericValue, "")
+              }
+            }
+          }
+        }
+      }
+    )
+
+    resourceValue.setFormats(formats)
+    resourceValue.namespaceResolver = namespaceResolver
+    val visibility = getVisibility(ATTR, name)
+
+    // todo kzheng handle attrNamespace
+    return BasicResourceItem(type = ATTR, name, visibility, configuration, resourceValue)
+  }
+
+  @Throws(
+    IOException::class,
+    XmlPullParserException::class,
+    XmlSyntaxException::class
+  )
+  private fun createPluralsItem(
+    name: String,
+    file: PathString,
+    configuration: RepositoryConfiguration
+  ): BasicResourceItem {
+    val resourceValue = object : PluralsResourceValueImpl(namespace, name, null, null) {
+      // Allow the user to specify a specific quantity to use via tools:quantity
+      override fun getValue(): String {
+        val quantity = parser.getAttributeValue(
+          ATTR_QUANTITY,
+          TOOLS_URI
+        )
+        if (quantity != null) {
+          val value = getValue(quantity)
+          if (value != null) {
+            return value
+          }
+        }
+        return super.getValue()
+      }
+    }
+    val defaultQuantity = parser.getAttributeValue(TOOLS_URI, ATTR_QUANTITY)
+    val namespaceResolver = parser.namespaceResolver
+    forSubTags(
+      TAG_ITEM,
+      object : XmlTagVisitor {
+        override fun visitTag() {
```

Comment:
for each:
```
    forSubTags(
      FOO,
      object : XmlTagVisitor {
        override fun visitTag() {
```

we can instead do:
```
    forSubTags(FOO) {
```

since `XmlTagVisitor` is a `@Functionalnterface`